synapse/events/third_party_rules.py

```diff
@@ -353,6 +365,50 @@ async def on_new_event(self, event_id: str) -> None:
                     "Failed to run module API callback %s: %s", callback, e
                 )
 
+    async def check_can_shutdown_room(self, user_id: str, room_id: str) -> bool:
+        """Intercept requests to shutdown a room. If `False` is returned, the
+        room should not be shut down.
+
+        Args:
+            requester: The ID of the user requesting the shutdown.
+            room_id: The ID of the room.
+
+        Raises:
+            ModuleFailureError if a callback raised any exception.
```

Comment:
It doesn't do that currently. Looks like the dosctring you used for inspiration is lying.